src/Model/PWLocal.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace Paymentwall\Paymentwall\Model;
+
+class PWLocal
+{
+    const STATE_PENDING_PAYMENT = 'pending_payment';
+    const PAYMENT_METHOD = 'paymentwall';
+
+    public function __construct($resultPageFactory, $storeManager, $product, $formkey, $quote, $quoteManagement,
```

Comment:
Please use type hinting and refactor this function.
It will help to read and coding easier.